modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/daemons/fetcher/YouTubeFetcher.scala

```diff
@@ -0,0 +1,149 @@
+      .withOptionQueryParam("publishedAfter", formattedPublishAfter)
+      .withOptionQueryParam("pageToken", pageToken)))(jsonOf[F, SearchListResponse]).flatMap { response =>
+
+      fetchVideoDetails(response.items.map(_.id.videoId)).flatMap { videos =>
+        val updatedVideos = result ++ videos.filter(v => v.duration > minimumDuration && v.viewCount > minimumViews)
+
+        response.nextPageToken match {
+          case Some(token) => fetchVideosByChannelId(
+            channelId,
+            searchString,
+            minimumDuration,
+            minimumViews,
+            publishedAfter,
+            Some(token),
+            updatedVideos
+          )
+          case None => updatedVideos.sortBy(_.publishedAt).pure
+        }
+      }
+    }
+  }
+
+  private def fetchVideoDetails(videoIds: List[String]): F[List[VideoDetails]] =
+    if (videoIds.isEmpty) Async[F].pure(Nil)
+    else client.expect[VideoListResponse](Request[F](Method.GET, Uri.unsafeFromString(baseUrl.toString() + "/videos")
+      .withQueryParam("key", apiKey)
+      .withQueryParam("id", videoIds.mkString(","))
+      .withQueryParam("part", "snippet,contentDetails,statistics")))(jsonOf[F, VideoListResponse]).flatMap { response =>
+      response.items.map(item =>
+        VideoDetails(
+          item.id,
+          item.statistics.viewCount,
+          Duration.parse(item.contentDetails.duration).getSeconds,
+          item.snippet.publishedAt
+        )
+      ).pure
+    }
+}
+
+object YouTubeFetcher {
+  def make[F[_] : Async : Network](
+    applicationConfig: ApplicationConfig,
+    calculatedStateService: CalculatedStateService[F]
+  ): Fetcher[F] = (currentDate: LocalDateTime) =>
+    MkHttpClient.forAsync[F].newEmber(applicationConfig.http4s.client).use { implicit client =>
+      val config = applicationConfig.youtubeDaemon
+
+      for {
+        latticeApiUrl <- config.usersSourceApiUrl.toOptionT.getOrRaise(new Exception(s"Could not get usersSourceApiUrl"))
+        baseUrl <- config.youtubeApiUrl.toOptionT.getOrRaise(new Exception(s"Could not get YouTube Data API baseUrl"))
+        apiKey <- config.youtubeApiKey.toOptionT.getOrRaise(new Exception(s"Could not get YouTube apiKey"))
+        searchInformation = config.searchInformation
+        youtubeFetcher = new YouTubeFetcher[F](apiKey, baseUrl)
+        dagUsers <- youtubeFetcher.fetchDagUsers(latticeApiUrl)
+        calculatedState <- calculatedStateService.get
+        dataSource: YouTubeDataSource = calculatedState.state.dataSources
+          .get(DataSourceType.YouTube)
+          .collect { case ds: YouTubeDataSource => ds }
+          .getOrElse(YouTubeDataSource(Map.empty))
+
+        dagUsersToReward = dagUsers.filterNot { user =>
+          dataSource.existingWallets
+            .filter(_._2.rewardsReceivedToday >= searchInformation.maxPerDay)
+            .keys.toList.contains(user.primaryDagAddress.get)
+        }
+
+        returningRewards <- dagUsersToReward.traverse { user =>
```

Comment:
I don't think we should name rewards here, at this point it's simply data updates, the rewards will happen in a second moment